File: go/exercise41/name_sorter.go
package main

import (
	"fmt"
	"os"
	"slices"
	"strings"
)

const file = "names.txt"

func main() {
	data, err := os.ReadFile(file)
	if err != nil {
		fmt.Println(err)
		return
	}

	// trim the spaces at the start and end
	trimmed := strings.TrimSpace(string(data))

	// convert to string array
	names := strings.Split(string(trimmed), "\n")
	slices.Sort(names)

	fmt.Printf("Total of %d names\n", len(names))
	fmt.Println(strings.Repeat("-", 10))
	for _, name := range names {
		fmt.Println(name)
	}
}

File: javascript/name-sorting/name-sorter.js
import fs from "node:fs";

const contents = fs.readFileSync("./names.txt", "utf8").trim().split("\n");

contents.sort((a, b) => a.localeCompare(b));

console.log(`Total of ${contents.length} names`);
console.log("-".repeat(18));
for (const name of contents) {
	console.log(name);
}
